### ISSUE TITLE
Possibly wrong constant values on any(target_os = "teeos", all(target_os = "linux", target_env = "uclibc", target_arch = "x86_64"))

### ISSUE THREAD

**Issue by ramhugger** [rust-lang/libc#3532]
Greetings,
I may have found a small set of constants whose values were included in [libc](https://github.com/rust-lang/libc) as decimal literals while in fact they were written as octal literals in C. I believe this is the case because in C an integer literal that begins with a '0' is considered an octal literal, whereas in Rust the leading '0' is silently discarded.

I first noticed this while inspecting the value of `O_NONBLOCK` of another platform for a project of mine, and I noticed it had [this declaration on Linux uClibc x86_64](https://github.com/rust-lang/libc/blob/1abc2e47107e03eab1757b8dcb24b1095f6aeeb9/src/unix/linux_like/linux/uclibc/x86_64/mod.rs#L309), which was inconsistent with all the other related values. This convinced me to investigate further, and I ran a RegExp against libc's codebase, which resulted in a small subset of constants (most are related to `fcntl.h`):
<img width="289" alt="image" src="https://github.com/rust-lang/libc/assets/37934707/671ab1bb-f17d-4067-920a-7eb49e670388">

Before filing this issue, in order to be as sure and detailed as possible, I also went and gave a read to [uClibc's Linux `fcntl.h` file](https://github.com/gittup/uClibc/blob/9dbf00be840a15a52656a039f83d1997344ce507/libc/sysdeps/linux/x86_64/bits/fcntl.h), which uses in fact octal literals to declare the affected constants' values.

## TL;DR

The constants listed above were likely copy-pasted from their respective headers and the subtle differences in integer literal rules from Rust and C lead to "wrong" values sneaking into the library...
unless I missed something and `rustc` is able to correctly interpret those literals as octals, which I unfortunately haven't been able to test on a Linux machine at the time of writing.

**Comment by tgross35** [comment]
Thanks for the catch! Could you put up a PR?

Also, do you have a link to the relevant teeos headers / docs?

libc location of the initial example https://github.com/rust-lang/libc/blob/1ffe0bde929a43ce3d25a8f08b9f95bde8d3e5d8/src/teeos/mod.rs#L180
